bin/runtime/src/lib.rs

```diff
@@ -384,15 +352,105 @@ impl<I: From<u32>> ::frame_support::traits::Get<I> for SessionPeriod {
     }
 }
 
+impl pallet_elections::Config for Runtime {
+    type Event = Event;
+    type DataProvider = Staking;
+    type SessionPeriod = SessionPeriod;
+}
+
+impl pallet_randomness_collective_flip::Config for Runtime {}
+
+parameter_types! {
+    pub const Offset: u32 = 0;
+}
+
 impl pallet_session::Config for Runtime {
     type Event = Event;
     type ValidatorId = <Self as frame_system::Config>::AccountId;
-    type ValidatorIdOf = ConvertInto;
+    type ValidatorIdOf = pallet_staking::StashOf<Self>;
     type ShouldEndSession = pallet_session::PeriodicSessions<SessionPeriod, Offset>;
     type NextSessionRotation = pallet_session::PeriodicSessions<SessionPeriod, Offset>;
-    type SessionManager = pallet_aleph::AlephSessionManager<Self>;
+    type SessionManager = pallet_session::historical::NoteHistoricalRoot<Self, Staking>;
     type SessionHandler = <SessionKeys as OpaqueKeys>::KeyTypeIdProviders;
     type Keys = SessionKeys;
+    type WeightInfo = pallet_session::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_session::historical::Config for Runtime {
+    type FullIdentification = pallet_staking::Exposure<AccountId, Balance>;
+    type FullIdentificationOf = pallet_staking::ExposureOf<Runtime>;
+}
+
+parameter_types! {
+    pub const BondingDuration: EraIndex = 14;
+    pub const SlashDeferDuration: EraIndex = 13;
+    pub const MaxNominatorRewardedPerValidator: u32 = 512;
+    pub const OffendingValidatorsThreshold: Perbill = Perbill::from_percent(33);
+    pub const DisabledValidatorsThreshold: Perbill = Perbill::from_percent(30);
+    pub const SessionsPerEra: EraIndex = DEFAULT_SESSIONS_PER_ERA;
+}
+
+const YEARLY_INFLATION: Balance = 30 * 1_000_000 * 1_000_000_000_000;
+// Milliseconds per year for the Julian year (365.25 days).
+const MILLISECONDS_PER_YEAR: u64 = 1000 * 3600 * 24 * 36525 / 100;
+
+pub struct UniformEraPayout {}
+
+impl pallet_staking::EraPayout<Balance> for UniformEraPayout {
+    fn era_payout(_: Balance, _: Balance, _: u64) -> (Balance, Balance) {
+        let miliseconds_per_era =
+            MillisecsPerBlock::get() * SessionPeriod::get() as u64 * SessionsPerEra::get() as u64;
+        let portion = Perbill::from_rational(miliseconds_per_era, MILLISECONDS_PER_YEAR);
+        let total_payout = portion * YEARLY_INFLATION;
+        let validators_payout = Perbill::from_percent(90) * total_payout;
+        let rest = total_payout - validators_payout;
+
+        (validators_payout, rest)
+    }
+}
```

Comment:
I think you might've wanted to use what got defined in `primitives` here? At least up to the two `const`s above.

Reply:
This is a split glitch.

Reply:
Fixed.